src/amm.rs

```diff
@@ -56,6 +56,10 @@ macro_rules! to_u128 {
 pub fn calculate_spot_price(sell_reserve: Balance, buy_reserve: Balance, amount: Balance) -> Option<Balance> {
     ensure!(sell_reserve != 0);
 
+    if amount == 0 || buy_reserve == 0 {
```

Comment:
Here again the formula is:

/// Formula : BUY_RESERVE * AMOUNT / SELL_RESERVE

So if either amount or buy_reserve are zero, there is no need to compute anything.
Does this make sense here?